// o2-upload-to-fbs/dist/src/o2-upload-to-fbs.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var angularfire2_1 = require("angularfire2");
var O2UploadToFbsComponent = (function () {
    function O2UploadToFbsComponent(firebaseApp) {
        this.storageRef = firebaseApp.storage().ref();
        this.flgShowMessage = true;
    }
    O2UploadToFbsComponent.prototype.ngOnInit = function () {
        if (this.btnSelectText == '') {
            this.btnSelectText = "Select File";
        }
        if (this.btnExecuteText == '') {
            this.btnSelectText = "Upload";
        }
    };
    O2UploadToFbsComponent.prototype.onBtnBrowseClick = function (event) {
        var uploadButton = document.getElementById("btnUpload");
        uploadButton.click();
    };
    O2UploadToFbsComponent.prototype.onBtnUploadChange = function (event) {
        var targetFile = event.srcElement.files[0];
        var uploader = document.getElementById("btnUpload");
        var fbsPath = this.fbsBasePath + targetFile.name;
        this.fileUploadFileName = targetFile.name;
        this.fileUploadFile = targetFile;
    };
    O2UploadToFbsComponent.prototype.onBtnExecuteClick = function (event) {
        var fbsPath = this.fbsBasePath + this.fileUploadFileName;
        this.uploadFile(fbsPath, this.fileUploadFile);
    };
    O2UploadToFbsComponent.prototype.uploadFile = function (fbsPath, targetFile) {
        var _this = this;
        var promise = new Promise(function (res, rej) {
            _this.targetRef = _this.storageRef.child(fbsPath);
            var task = _this.targetRef.put(targetFile);
            task.on('state_changed', function (snapshot) {
                console.log(snapshot.state);
            }, function (error) {
                console.log(error.code);
                if (_this.flgShowMessage) {
                    var msg = "Error " + error.code;
                    window.alert(msg);
                }
                rej(error);
            }, function () {
                var downloadUrl = task.snapshot.downloadURL;
                console.log(downloadUrl);
                if (_this.flgShowMessage) {
                    var msg = "Uploaded ---> " + fbsPath;
                    window.alert(msg);
                }
                res(downloadUrl);
            });
        });
        return promise;
    };
    return O2UploadToFbsComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], O2UploadToFbsComponent.prototype, "fbsBasePath", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], O2UploadToFbsComponent.prototype, "btnSelectText", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], O2UploadToFbsComponent.prototype, "btnExecuteText", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], O2UploadToFbsComponent.prototype, "inpPlaceholderText", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], O2UploadToFbsComponent.prototype, "flgShowMessage", void 0);
O2UploadToFbsComponent = __decorate([
    core_1.Component({
        selector: 'o2-upload-to-fbs',
        template: "\n  <input type=\"file\" id=\"btnUpload\" style=\"display:none;\"  (change)=\"onBtnUploadChange($event)\" />\n  <input [(ngModel)]=\"fileUploadFileName\" placeholder={{inpPlaceholderText}} Size=\"30\">\n  <button type=\"button\" (click) = \"onBtnBrowseClick($event)\">{{btnSelectText}}</button>\n  <button type=\"button\" (click) = \"onBtnExecuteClick($event)\">{{btnExecuteText}}</button>\n  "
        //   ,
        //   styleUrls: ['./o2-upload-to-fbs.component.css']
    }),
    __param(0, core_1.Inject(angularfire2_1.FirebaseApp)),
    __metadata("design:paramtypes", [Object])
], O2UploadToFbsComponent);
exports.O2UploadToFbsComponent = O2UploadToFbsComponent;
