// o2-upload-to-fbs/dist/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var angularfire2_1 = require("angularfire2");
var o2_upload_to_fbs_component_1 = require("./src/o2-upload-to-fbs.component");
__export(require("./src/o2-upload-to-fbs.component"));
var O2UploadToFbsModule = O2UploadToFbsModule_1 = (function () {
    function O2UploadToFbsModule() {
    }
    O2UploadToFbsModule.forRoot = function () {
        return {
            ngModule: O2UploadToFbsModule_1,
            providers: []
        };
    };
    return O2UploadToFbsModule;
}());
O2UploadToFbsModule = O2UploadToFbsModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule,
            angularfire2_1.AngularFireModule
        ],
        declarations: [
            o2_upload_to_fbs_component_1.O2UploadToFbsComponent,
        ],
        exports: [
            o2_upload_to_fbs_component_1.O2UploadToFbsComponent,
        ]
    })
], O2UploadToFbsModule);
exports.default = O2UploadToFbsModule;
var O2UploadToFbsModule_1;
